WordPress/Classes/ViewRelated/Me/Change Username/ChangeUsernameViewController.swift

```diff
@@ -0,0 +1,162 @@
+        let bottomInset = max(relativeRect.height - relativeRect.maxY + view.frame.height, 0)
+        var insets = tableView.contentInset
+        insets.bottom = bottomInset
+        tableView.contentInset = insets
+        tableView.scrollIndicatorInsets = insets
+    }
+
+    func save() {
+        present(changeUsernameConfirmationPrompt(), animated: true)
+    }
+
+    func changeUsername() {
+        SVProgressHUD.show()
+        viewModel.save() { [weak self] (state, error) in
+            switch state {
+            case .success:
+                SVProgressHUD.dismiss()
+                self?.completionBlock()
+                self?.navigationController?.popViewController(animated: true)
+            case .failure:
+                SVProgressHUD.showError(withStatus: error)
+            default:
+                break
+            }
+        }
+    }
+
+    func changeUsernameConfirmationPrompt() -> UIAlertController {
+        let alertController = UIAlertController(title: Constants.Alert.title,
+                                                message: "",
+                                                preferredStyle: .alert)
+        alertController.addAttributeMessage(String(format: Constants.Alert.message, viewModel.selectedUsername),
+                                            highlighted: viewModel.selectedUsername)
+        alertController.addTextField { textField in
+            textField.placeholder = Constants.Alert.confirm
+        }
+        alertController.addCancelActionWithTitle(Constants.Alert.cancel)
+        alertController.addDefaultActionWithTitle(Constants.Alert.change, handler: { [weak alertController] _ in
+            guard let textField = alertController?.textFields?.first,
+                textField.text == self.viewModel.selectedUsername else {
+                    DDLogInfo("Username confirmation failed")
+                    return
+            }
+            DDLogInfo("User changes username")
+            self.changeUsername()
+        })
+        DDLogInfo("Prompting user for confirmation of change username")
+        return alertController
+    }
+
+    enum Constants {
+        static let actionButtonTitle = NSLocalizedString("Save", comment: "Settings Text save button title")
+        static let username = NSLocalizedString("Username", comment: "The header and main title")
+        static let cellIdentifier = "SearchTableViewCell"
+
+        enum Alert {
+            static let loading = NSLocalizedString("Loading usernames", comment: "Shown while the app waits for the username suggestions web service to return during the site creation process.")
+            static let title = NSLocalizedString("Careful!", comment: "Alert title.")
+            static let message = NSLocalizedString("You are changing your Username to %@. Changing your username will also affect your Gravatar profile and IntenseDebate profile addresses. \nConfirm your new Username to continue.", comment: "Alert message.")
```

Comment:
Nitpick, but a couple of occurrences of 'username' in here have a capital U. I think they should all be lowercase?